### ISSUE TITLE
karma hydrate requires git executable ??

### ISSUE THREAD

**Issue by killerwolf** [Niktux/karma#82]
WHY ????

bin/karma hydrate -e dev settings/
PHP Fatal error:  Uncaught exception 'GitWrapper\GitException' with message 'Unable to find the Git executable.' in /var/www/mywebsite/current/application/vendor/cpliakas/git-wrapper/src/GitWrapper/GitWrapper.php:78

**Comment by Niktux** [comment]
Hello,

Karma should only need git when you use "vcs" command. After a fast glance, I think the issue comes from the DI container.

https://github.com/Niktux/karma/blob/89340943c7/src%2FKarma%2FApplication.php#L165

This line instanciates by error git objects. 
I can commit a fix in a few minutes but I've no server without git to test it at the moment.

If I commit the fix, can you test on your own ?

**Comment by Niktux** [comment]
You can test the fix with composer :

```json
"niktux/karma": "dev-82-wrong-dependency-to-git@dev"
```

**Comment by killerwolf** [comment]
Sure....hold on.

**Comment by killerwolf** [comment]
Tested....and it's working as expected :)

**Comment by killerwolf** [comment]
why not moving
```
"cpliakas/git-wrapper" : "~1.2",
```
from "require" to "suggest"

**Comment by Niktux** [comment]
Ok. Thanks.
I will produce a stable release tag (5.2.1) tonight

**Comment by Niktux** [comment]
It's a good question but the answer is not obvious. This package is not required by all the commands, that's true. But it's not optional : the need of this dependency is enabled by default because one of the core commands needs it. 

The correct behaviour would be to require or suggest "git" but it's out of scope of composer (https://getcomposer.org/doc/02-libraries.md#platform-packages)

**Comment by Niktux** [comment]
Fixed in 5.2.1
